flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  description = "A garnix module for Uptime Kuma";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
+  inputs.garnix-lib.url =
+    "github:jfroche/garnix-lib/jfroche/expose-module-for-system";
+
+  outputs = { self, nixpkgs, garnix-lib, }:
+    let
+      lib = nixpkgs.lib;
+      pkgs = import nixpkgs { system = "x86_64-linux"; };
+      uptimeKumaSubmodule.options = {
+        port = lib.mkOption {
+          type = lib.types.port;
+          description = "The port in which to run uptime-kuma";
+          default = 3001;
+        };
+        path = lib.mkOption {
+          type = lib.types.nonEmptyStr;
+          description = "Webserver path to host your uptime-kuma server on";
+          default = "/";
+        };
+      };
+    in {
+      garnixModules.default = { pkgs, config, ... }: {
+        options = {
+          uptimeKuma = lib.mkOption {
+            type = lib.types.attrsOf (lib.types.submodule uptimeKumaSubmodule);
+            description = "An attrset of uptime-kuma instances";
+          };
+        };
+
+        config = {
+          nixosConfigurations.default = builtins.attrValues (builtins.mapAttrs
+            (name: projectConfig: {
+              services.uptime-kuma = {
+                enable = true;
+                settings.UPTIME_KUMA_PORT = toString projectConfig.port;
+              };
+              garnix.server.persistence = {
+                enable = true;
+                name = "uptimeKuma";
```

Comment:
If a user enables this module and e.g. the `postgres` module, will this option setting clash? (To fix this might require some changes that are probably out of scope for this PR. But having an answer to the question would be nice.)

Reply:
I have set a higher priority to the `garnix.server.persistence.name` to avoid that name clash.

Reply:
Hmm, does that mean that if anyone added the uptimeKuma module later to an existing persistent server, they would loose the persistent state of that server, since this `name` would change?